addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -511,63 +511,50 @@ public synchronized Boolean isConnected() {
     }
 
     public boolean isOnline() {
-        return _online;
-    }
-
-    private synchronized boolean getConnected() {
-        return this.connected;
+        ThingStatusInfo statusInfo = getThing().getStatusInfo();
+        return (statusInfo.getStatus() == ThingStatus.ONLINE) ? true : false;
     }
 
     /**
      * Set connection status.
      *
      * @param connected
+     * @throws ZoneMinderUrlNotFoundException
+     * @throws IOException
+     * @throws GeneralSecurityException
+     * @throws IllegalArgumentException
      */
-    private synchronized void setConnected(boolean connected) {
-
+    private void setConnected(boolean connected)
+            throws IllegalArgumentException, GeneralSecurityException, IOException, ZoneMinderUrlNotFoundException {
         if (this.connected != connected) {
-            if (connected) {
-                try {
-                    zoneMinderSession = ZoneMinderFactory.CreateSession(zoneMinderConnection);
-                } catch (FailedLoginException | IllegalArgumentException | IOException
-                        | ZoneMinderUrlNotFoundException e) {
-                    logger.error("BRIDGE [{}]: Call to setConencted failed with exception '{}'", getThingId(),
-                            e.getMessage());
-                }
-            } else {
-                zoneMinderSession = null;
-            }
-            this.connected = connected;
-
-        }
+            try {
+                if (connected) {
+                    try {
+                        if (zoneMinderEventSession == null) {
+                            zoneMinderEventSession = ZoneMinderFactory.CreateEventSession(zoneMinderConnection);
+                        }
 
-    }
+                    } catch (Exception ex) {
```

Comment:
Please catch specific exceptions instead of catching all exceptions. That way it is more clear what exceptions need to be handled and how this is done.